app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/MessageService.scala

```diff
@@ -102,6 +104,36 @@ class MessageService @Inject() (
       .map(_.getOrElse(UpdateOutcome.Locked))
   }
 
+  def fixProblemMovement(id: String)(implicit hc: HeaderCarrier): Future[Done] =
+    movementRepository.getMovementById(id).flatMap {
+      _.map { movement =>
+        val movementWithNoMessages = movement.copy(messages = Seq.empty, administrativeReferenceCode = None)
+        movement.messages
+          .sortBy(_.createdOn)
```

Comment:
Is there something better we can do rather than chaining flat map and map after the fold left?